### DISCUSSION TITLE
Generating a re-sizable QR code with logo in the middle which can be transparent and may change colors.

### DISCUSSION THREAD

**Discussion by sukith** [chillerlan/php-qrcode#189] [Q&A]
Dear codemasher,

Highly appreciate for creating this extensive library for us! I have faced the following issues with it. Below is my requirement.

I want to create a QR code that is dynamically able to

1.  Change the QR image size in pixels
2. Able to have a logo in the middle - with some margin with the QR code without overlapping
3.  Able to be transparent - without removing white space from the logo background
4. Able to change foreground and background colors when required.
5. Have a quiet space size of 2 

When I tried some earlier tutorials from you and the community I could perform these tasks individually but couldn't create an all-in-one solution. I got issues in different versions and errors that I couldn't debug. Will you be able to quickly direct me toward generating this?

**Comment by codemasher** [comment]
Hi! The [imageWithLogo example](https://github.com/chillerlan/php-qrcode/blob/02a2321dbbac541abc3839a7b27ccdb222e7cd4a/examples/imageWithLogo.php) does pretty much all of this - the only thing you would need to do is to resize the final output to whatever size you need.

> Change the QR image size in pixels

adjust the size of the output via `QROptions:$scale`
https://github.com/chillerlan/php-qrcode/blob/02a2321dbbac541abc3839a7b27ccdb222e7cd4a/examples/imageWithLogo.php#L84


> Able to have a logo in the middle - with some margin with the QR code without overlapping

The example does exactly that over here:
https://github.com/chillerlan/php-qrcode/blob/02a2321dbbac541abc3839a7b27ccdb222e7cd4a/examples/imageWithLogo.php#L50-L52


> Able to be transparent - without removing white space from the logo background

Set  `QROptions:$imageTransparent` to `true`, add the options `QROptions:$bgColor` and  `QROptions:$transparencyColor` and set the values for both of these to a color other than the white (e.g. `[0, 255, 0]`) - this color will be set to transparent via [`imagecolortransparent()`](https://www.php.net/manual/function.imagecolortransparent).
https://github.com/chillerlan/php-qrcode/blob/02a2321dbbac541abc3839a7b27ccdb222e7cd4a/examples/image.php#L22-L24


> Able to change foreground and background colors when required.

See above and the  [simple image example](https://github.com/chillerlan/php-qrcode/blob/02a2321dbbac541abc3839a7b27ccdb222e7cd4a/examples/image.php).


> Have a quiet space size of 2

Set `QROptions:$quietzoneSize` to `2`.

Hope this helps!
